cleanlab/internal/label_quality_utils.py

```diff
@@ -95,21 +95,32 @@ def get_normalized_entropy(
 
     Parameters
     ----------
-    pred_probs:
+    pred_probs : np.ndarray (shape (N, K))
       Each row of this matrix corresponds to an example x and contains the model-predicted
       probabilities that x belongs to each possible class: P(label=k|x)
 
-    min_allowed_prob:
-      Minimum allowed probability value. Entries of `pred_probs` below this value will be clipped to this value.
-      Ensures entropy remains well-behaved even when `pred_probs` contains zeros.
+    min_allowed_prob : float, default: None, deprecated
+      Minimum allowed probability value. If not `None` (default),
+      entries of `pred_probs` below this value will be clipped to this value.
+
+      .. deprecated:: 2.5.0
+         This keyword is deprecated and should be left to the default.
+         The Entropy is well-behaved even if `pred_probs` contains zeros,
+         clipping is unnecessary and (slightly) changes the results.
 
     Returns
     -------
-    entropy:
+    entropy : np.ndarray (shape (N, ))
       Each element is the normalized entropy of the corresponding row of ``pred_probs``.
     """
     num_classes = pred_probs.shape[1]
 
+    if min_allowed_prob is not None:
+        warnings.warn(
+            "Using `min_allowed_prob` is not necessary anymore and will be removed.",
+            DeprecationWarning,
+        )
+        pred_probs = np.clip(pred_probs, a_min=min_allowed_prob, a_max=None)
+
     # Note that dividing by log(num_classes) changes the base of the log which rescales entropy to 0-1 range
-    clipped_pred_probs = np.clip(pred_probs, a_min=min_allowed_prob, a_max=None)
-    return -np.sum(pred_probs * np.log(clipped_pred_probs), axis=1) / np.log(num_classes)
+    return -np.sum(xlogy(pred_probs, pred_probs), axis=1) / np.log(num_classes)
```

Comment:
You are right, this might be breaking behavior. Before, negative probabilities were silently ignored, as they were clipped to a tiny positive value.

Instead of an `assert`, I would rather raise a `ValueError` which seems suitable in this case. Assertion are not thrown if optimizations are turned on, and typically don't belong in production code.
The question follow-up question is, whether we should also catch probabilities larger than zero. While the formula is mathematically still well-defined, its meaning is questionable.

Another option would be simply rising a `RuntimeWarning` as `numpy.log` does for negative numbers (actually, I assumed that `xlogy` would behave the same way, but I forgot to check it).

Reply:
Yes a `ValueError` sounds good. I think adding a check to ensure that probabilities are not larger than 1 is a good idea as well, good catch there!